### DISCUSSION TITLE
Edit button question

### DISCUSSION THREAD

**Discussion by RaduSeba** [Arquisoft/lomap_en3b#70] [Ideas]
## Do we need the edit button?

![idea](https://user-images.githubusercontent.com/98213239/232571600-34217511-548b-49fd-85d4-bb3f045240ae.png)

- I does't say nothing in the requirement about it and I think that if we edit a location we should also delete it s reviews.
- I think If we want a button like this it s better to add a simple delete button that removes a location. So than you can add another one instead.

**Comment by FdezAriasSara** [comment]
Mmm you're right it does not say anything about it in the requirements...

Since it's already implemented , what i suggest is to move the edit option from the header to the info window that appears when you click a location :
![image](https://user-images.githubusercontent.com/79209608/232719614-748c9536-953c-4058-8bec-2ed69874345b.png)

or in the review submenu something like:
![image](https://user-images.githubusercontent.com/79209608/232720303-2c258699-ac91-4bce-b28c-3e4627d3c9ff.png)

(i suggest that both re-direct the user to the already existing modal dialogue) 



in both cases i suggest to change the edit icon to something like this, since it's more conventional and therefore it improves usability: 
![image](https://user-images.githubusercontent.com/79209608/232719784-d8346ba9-3a94-4878-8238-5f116e2b52b0.png)


anyways, since it's optional  i would leave it  until the main requirements are fully done and then we can decide if we have time to enhance it or its better to delete it

**Reply by FdezAriasSara** [reply]
Also , while saying all of that im guessing it's "easy" to make once the insert-location in the pod is fully working, but im not 100% sure if that is actually like that 😵‍💫

**Reply by RaduSeba** [reply]
Yes, I think it s better to leave the edit button after we have the minimum requirements

**Comment by FdezAriasSara** [comment]
By the way, these two buttons
![image](https://user-images.githubusercontent.com/79209608/232856730-e7c1c0f7-564d-4935-95f8-9d652ff38f34.png)
they do the same right? 


@RaduSeba we agreed on creating new locations by making a click on the map, so , wouldnt these buttons be unnecesary?
The only idea i have where i see the use of an add location button would be to allow the user to create a location with exact coordinates

In case we did that, we still would have to delete one of the icons i think..



+ here
![image](https://user-images.githubusercontent.com/79209608/232857554-a38ea2b4-25ed-4839-8ab3-5f2fd27c728b.png)

map icon button seems to be doing nothing too

Let me know what you think! :) 
@RaduSeba @Batuhanbyr

**Reply by RaduSeba** [reply]
So as it is now if you want to add a location you need to click on the icon with the marker and the plus and then a form will pop up and after you complete it you click on the map too add a location.

Yes two buttons is too much. We need only one. Maybe the button on the right can be switched to a logout button.

I don't know what the Map icon does. I think we should delete it.

We can also do the add button bigger or position it somewhere else so it is easier to figure out where it is and what it does.

**Comment by Batuhanbyr** [comment]
### Edit Location Button 
- I agree with @RaduSeba . I don't think we need a button or field to edit location information, as users can simply delete and re-add their location. So, we can delete this button and EditList.js.
###  Plus Button 
- I also agree with @RaduSeba about this button. I can turn this button into something containing a logout pop up.
### Map Button 
- I have a lot of questions about this button. I think it should be like the Friends button, where we can see a list of our friends and when we click on any of them, we should go to their own map. This way, we can see the locations our friends have added. Otherwise, how will we access a location that our friends have added and how will we add comments to that location? 
### Edit Button inside InfoWindow
- Having an edit button inside the infowindow can create a few issues, as I can also look at a location added by my friend and change it even if I didn't add that location. So, if I didn't add that location, this button should be disabled so that only the creator of that location can change it.
### Home Button
- I think we should make something like a window that appears when we click on the button with  profile Icon, but it should cover the entire page. What do you think about this button? Should we delete it or keep it?

### so what we need to do? @RaduSeba @FdezAriasSara

**Comment by Batuhanbyr** [comment]
- I can make the logout button as a dialog
[material uı dialog](https://mui.com/material-ui/react-dialog/)

- ![mapicon](https://user-images.githubusercontent.com/95686987/232879133-6ff8c854-416f-4efe-a9bd-85a5305bccaf.jpg)

- Alternatively, we can add a button inside the Friends section to go to our friend's map.
